app/services/foods_api/client.rb

```diff
@@ -0,0 +1,40 @@
+require_relative 'endpoints'
+require_relative 'api_calls'
+
+module FoodsAPI
+  class Client
+    extend Endpoints
+    extend ApiCalls
+
+    def self.client(params)
+      params[:api_key] = ENV['API_KEY']
+
+      Faraday.new(
+        url: api_endpoint,
+        params: params
+      )
+    end
+
+    def self.get(endpoint = nil, params: {})
+      raise 'API endpoint must be defined' if endpoint.nil?
+
+      connection = client(params)
+
+      @response = connection.get(endpoint)
+
+      return parse_json if response_successful?
+
+      raise "Status: #{@response.status}, Response: #{@response.body}"
+    end
```

Comment:
for these challenges, we're not expecting you to worry much about these edge cases that you have in here.